File: Puntonet/Puntonet.Web/Modules/ServerTypes/Persons/PersonsForm.ts
import { StringEditor, EnumEditor, LookupEditor, PrefixedContext } from "@serenity-is/corelib";
import { Gender } from "../Web/Modules.Persons.Persons.Gender";
import { AccesoriesDialog } from "@/Parameters/Accesories/AccesoriesDialog";
import { PersonClothesEditor } from "@/Parameters/PersonClothes/PersonClothesEditor";
import { PersonAtributtesEditor } from "@/Parameters/PersonAtributtes/PersonAtributtesEditor";
import { initFormType } from "@serenity-is/corelib/q";

export interface PersonsForm {
    Name: StringEditor;
    Address: StringEditor;
    Phone: StringEditor;
    Identity: StringEditor;
    Gender: EnumEditor;
    Photo: StringEditor;
    AccesoriList: LookupEditor;
    ClothesList: PersonClothesEditor;
    Atributtes: PersonAtributtesEditor;
}

export class PersonsForm extends PrefixedContext {
    static formKey = 'Persons.Persons';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PersonsForm.init)  {
            PersonsForm.init = true;

            var w0 = StringEditor;
            var w1 = EnumEditor;
            var w2 = LookupEditor;
            var w3 = PersonClothesEditor;
            var w4 = PersonAtributtesEditor;

            initFormType(PersonsForm, [
                'Name', w0,
                'Address', w0,
                'Phone', w0,
                'Identity', w0,
                'Gender', w1,
                'Photo', w0,
                'AccesoriList', w2,
                'ClothesList', w3,
                'Atributtes', w4
            ]);
        }
    }
}

[Gender, AccesoriesDialog]; // referenced types

File: Puntonet/Puntonet.Web/Modules/Persons/Persons/PersonsForm.cs
using Org.BouncyCastle.Asn1.X509.SigI;
using Puntonet.Parameters;
using Puntonet.Persons.PersonAtributtes;
using Puntonet.Web.Modules.Persons.Persons;
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Web;
using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace Puntonet.Persons.Forms
{
    [FormScript("Persons.Persons")]
    [BasedOnRow(typeof(PersonsRow), CheckNames = true)]
    public class PersonsForm
    {
        [Tab("Personal Data")]
        [HalfWidth]
        public string Name { get; set; }
        [HalfWidth]
        public string Address { get; set; }
        [HalfWidth]
        public string Phone { get; set; }
        [HalfWidth]
        public string Identity { get; set; }
        [HalfWidth]
        [DefaultValue("1")]
        public Gender Gender { get; set; }
        [HalfWidth]
        public string Photo { get; set; }
        [Tab("Extras")]
        public ListField<Int32> AccesoriList { get; set; }
        [PersonClothesEditor]
        public List<PersonClothesRow> ClothesList { get; set; }
        [PersonAtributtesEditor]
        public List<PersonAtributtesRow> Atributtes { get; set; }

    }
}